### PR TITLE
Ios webview

### PR DESCRIPTION
**Pull request by minggo**

### CODE REVIEW

**Comment by minggo** [comment]
Replace UIWebView with WKWebView.

File: cocos/ui/UIWebViewImpl-ios.mm [hunk 1]
```diff
@@ -28,6 +28,10 @@ of this software and associated documentation files (the "Software"), to deal
 // Webview not available on tvOS
 #if (CC_TARGET_PLATFORM == CC_PLATFORM_IOS) && !defined(CC_TARGET_OS_TVOS)
 
+#import <WebKit/WKWebView.h>
+#import <WebKit/WKUIDelegate.h>
+#import <WebKit/WKNavigationDelegate.h>
+
```
Comment (cylemonVip) [inline cocos/ui/UIWebViewImpl-ios.mm:34]: 此处应该引入 #import <WebKit/WKNavigationAction.h>  否则 WKNavigationAction 对象无法引用其变量

File: cocos/ui/UIWebViewImpl-ios.mm [hunk 2]
```diff
@@ -211,83 +201,107 @@ - (void)loadUrl:(const std::string &)urlString cleanCachedData:(BOOL) needCleanC
...
 
 
-#pragma mark - UIWebViewDelegate
-- (BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)request navigationType:(UIWebViewNavigationType)navigationType {
-    NSString *url = [[request URL] absoluteString];
-    if ([[[request URL] scheme] isEqualToString:self.jsScheme]) {
+#pragma mark - WKNavigationDelegate
+- (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(WKNavigationAction *)navigationAction decisionHandler:(void (^)(WKNavigationActionPolicy))decisionHandler {
+    NSString *url = [webView.URL absoluteString];
+    if ([[webView.URL scheme] isEqualToString:self.jsScheme]) {
```
Comment (cylemonVip) [inline cocos/ui/UIWebViewImpl-ios.mm:254]: 此处 webView.URL 应该改为 navigationAction.request.URL 因为是决定是否重定向，应该用请求的url，而不是当前url

Comment (minggo) [reply cocos/ui/UIWebViewImpl-ios.mm:254]: It is fixed. Please use english in future.

File: cocos/ui/UIWebViewImpl-ios.mm (hunk 1 above)
Comment (minggo) [reply cocos/ui/UIWebViewImpl-ios.mm:34]: How to reproduce the issue?

**Comment by AIGRIND-LLC** [comment]
@minggo it seems `setBackgroundTransparent` does not work in new implementation.
Should I create a dedicated issue?

**Comment by minggo** [comment]
Yes, please. And please describe how to reproduce it.

**Pull request merged**